### ISSUE TITLE
[Go] Implement new Concept Exercise: time-duration

### ISSUE THREAD

**Issue by oanaOM** [exercism/v3#2072]
This issue describes how to implement the `time-duration` concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how concept time and duration is implemented in Go. 

## Learning objectives

- Define the relation between time.Time and time.Duration.
- Know how to measure time using time.Since ( e.g. execution time of a function -- no `defer` !) .
- Know how to use `time.Add` and `time.Sub`
- Know about time units and maybe do some calculations
- How to use time.Sleep (?)

## Out of scope

- Measuring time should to include `defer` 

## Concepts

- `time-duration`

## Prerequisites

- `time`

### Hints

- [Time package](https://golang.org/pkg/time/)

## Analyzer

This exercise does not require any specific logic to be added to the analyzer

## Implementing

To implement this exercise, please follow these [instructions](https://github.com/exercism/v3/blob/master/languages/go/reference/implementing-a-concept-exercise.md) and the below file structure.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by SourangshuGhosh** [comment]
@oanaOM can I take this issue. I want to do my first contribution to get started with doing mentorship in exercism website

**Comment by oanaOM** [comment]
Great! It's yours @SourangshuGhosh !
